### ISSUE TITLE
Popcorn files loader, as implemented, isn't caching files

### ISSUE THREAD

**Issue by JohnSmith-LT** [thexerteproject/xerteonlinetoolkits#1105]
The Popcorn files loader, as implemented in the Interactive Video page is using $.getScript which does an ajax load of each of the required js files. The problem is that the .js files have been cache busted and therefore load fresh each time an LO with that (and possibly other pages) load. At least they don't appear to reload if you add several IV pages to the same LO but I think, given that they are nearly 0.5Mb that we should try to cache where possible, at very least for promoting mobile use (the first image is normal broadband - the second is Fast 3G throttling in Chrome)

![image](https://user-images.githubusercontent.com/3907362/171611462-260573e3-230a-4747-99ad-6037f9670311.png)
![image](https://user-images.githubusercontent.com/3907362/171614108-d5273b05-4472-4da5-ad8f-0f4159a1c1e8.png)

There was a $script dependency loader added to Xenith a number of years ago and it is in use in several places but I have recently removed this and replaced with a more up-to-date one, based on $script but supporting Promises, which has decent browser compatibility (https://github.com/muicss/loadjs). More importantly though it supports success and failure detection and also supports loading of CSS and image files as well as asynchronous loading (can download files in parallel but load in order presented). There are other features such as bundling and you can then separate you Load code from your Detect and Execute code which helps pages load faster.

The loader in Interactive Video can then be rewritten as (using the original arrays):

            this.loadJS = function() {
                loadjs.path(x_templateLocation + 'common_html5/js/popcorn/');
                loadjs([...plugins.map(j => 'plugins/popcorn.' + j + '.js'), ...css.map(c => 'plugins/popcorn.' + c + '.css'), 'popcorn-complete.js'], {
                    success: function() {
                        interactiveVideo.setUp();
                    },
                    error: function(failed) {
                        console.log("Failed to load Popcorn.js & plugins: ", failed);
                    },
                    async: false // Fetch files in parallel but load in series
                });
            };

[ ignore the ... spread operator for now - that needs rewritten for backward compatibility at present]

Also I notice that, despite your best efforts to force popcorn-complete.js to load last that using the original loader it actually loads FIRST! In this case Popcorn obviously has checks built into accomodate this but in some situations that could be a deal breaker...

![image](https://user-images.githubusercontent.com/3907362/171612512-3f4f3dc0-25f3-4886-bafb-f107269ac97c.png)

Is this going to seriously affect the work that you have been doing @torinfo to merge in the Popcorn library? It shouldn't be too onerous to rewrite any of these loaders using the new method (if you agree that it is better). Also, it's handier for debugging purposes since it will give you an array of the files that don't load:

![image](https://user-images.githubusercontent.com/3907362/171612880-f3323fda-7181-4cae-a9f4-d2f5eb354201.png)

These could be used to target a CDN or similar alternative source for fall-over protection...

Should I commit this change to Interactive Video to develop for you to look at @torinfo ? Also is it worth rewriting to remove spread operator for IE11 support given that it ends officially this month? Happy to do it but might be pointless...

**Comment by JohnSmith-LT** [comment]
AdaptiveContent.html could also benefit from rewriting in this way also https://github.com/thexerteproject/xerteonlinetoolkits/blob/db03c12854de11be69f8739fa17e44d48e1a245b/modules/xerte/parent_templates/Nottingham/models_html5/adaptiveContent.html#L680

**Comment by torinfo** [comment]
@JohnSmith-LT I agree this is better. I didn't know about the central functionality. This also holds for mediaLesson and Adaptive content, and probably others as well. I will take care of the merge and implementation in other pages that use popcorn now (in my fork)

**Comment by JohnSmith-LT** [comment]
Hi @torinfo 

Just now we have 4x page models that use  the loadJS dependency loader:

- [ ] - flashCards.html
- [ ] - opinion.html
- [ ] - QRcode.html
- [ ] - wordsearch.html

but still have 11x files that use $.getScript still:

- [ ] - adaptiveContent.html
- [ ] - interactiveVideo.html
- [ ] - media360.html
- [ ] - mediaLesson.html
- [ ] - nav.html
- [ ] - perspectives.html
- [ ] - showGraph.html
- [ ] - SictTimeline.html
- [ ] - textVideo.html
- [ ] - videoSynch.html
- [ ] - youtube.html

as well as:

- [ ] - xenith.js,
- [ ] - application.js,
- [ ] - popcorn.HTMLMediasiteVideoElement.js

Are you ok with me updating these to use the new loader now to stop the popcorn and other files being cache busted? Just want to make sure you didn't already do this and still to commit?

**Comment by torinfo** [comment]
@JohnSmith-LT No, I did not have the time yet to do this, so yes, please!
